stetho/src/main/java/com/facebook/stetho/inspector/screencast/ScreencastDispatcher.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.screencast;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.RectF;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Looper;
+import android.util.Base64;
+import android.util.Base64OutputStream;
+import android.view.View;
+import com.facebook.stetho.common.LogUtil;
+import com.facebook.stetho.inspector.elements.android.ActivityTracker;
+import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
+import com.facebook.stetho.inspector.protocol.module.Page;
+
+import java.io.ByteArrayOutputStream;
+
+public final class ScreencastDispatcher {
+  private static final float FRAMERATE = 60f;
+  private static final long FRAME_DELAY = (long) (1000 / FRAMERATE);
+
+  private final Handler mMainHandler = new Handler(Looper.getMainLooper());
+  private final BitmapFetchRunnable mBitmapFetchRunnable = new BitmapFetchRunnable();
+  private final ActivityTracker mActivityTracker = ActivityTracker.get();
+  private final EventDispatchRunnable mEventDispatchRunnable = new EventDispatchRunnable();
+
+  private boolean mIsRunning;
+  private Handler mBackgroundHandler;
```

Comment:
actually nothing is initialized in the constructor (it's currently empty). Stuff only gets allocated inside `startScreencast` thus this can't be final